pytest_mergify/__init__.py

```diff
@@ -164,6 +226,20 @@ def pytest_runtest_protocol(
             else:
                 skip_attributes = {}
 
+                if (
+                    item.nodeid in self.quarantined_tests
+                    and item.get_closest_marker("xfail") is None
+                ):
+                    item.add_marker(
+                        pytest.mark.xfail(
+                            reason="Marked as xfail(non-strict) because the test is quarantined",
```

Comment:
I think it could be worth focusing on the why more than on the how :) 
```suggestion
                            reason="Test is quarantined by Mergify CI-Insights",
```